app/src/main/java/dev/candar/swing/JFFile.java

```diff
@@ -30,17 +29,26 @@ class JFFile extends JPanel {
         gc.weightx = 1;
         gc.fill = GridBagConstraints.HORIZONTAL;
 
-        add(new JLabel( 
-            new ImageIcon(Utils.iconToImage(isDirectory ? Resources.dirIcon : Resources.fileIcon)
-                .getScaledInstance(50, 50, Image.SCALE_FAST))
-        ), gc);
+        JLabel icon = new JLabel( 
+            (dim ?
+                isDirectory ? Resources.folderCutImageIcon : Resources.fileCutImageIcon :
+                isDirectory ? Resources.folderImageIcon : Resources.fileImageIcon)
+        );
```

Comment:
Nitpick: Multiple line ternary operators are not preferred. Using if-else statements would be better for readability.